torch/csrc/autograd/profiler.cpp

```diff
@@ -46,295 +46,319 @@ namespace {
+      std::lock_guard<std::mutex> g(state_mutex_);
+      thread_event_lists result;
+      for (auto& kv : event_lists_map_) {
+        auto& list = kv.second;
+        result.emplace_back(list->consolidate());
+      }
+      // Consolidate remote events if applicable as well.
+      if (remoteProfiledEvents_) {
+        result.insert(
+            result.end(),
+            std::make_move_iterator(remoteProfiledEvents_->begin()),
+            std::make_move_iterator(remoteProfiledEvents_->end()));
+      }
+      return result;
     }
-  }
 
-  void setOrAddRemoteProfiledEvents(std::vector<Event>&& remoteProfiledEvents) {
-    // Lock to serialize access from multiple callback threads.
-    std::lock_guard<std::mutex> guard(state_mutex_);
-    if (remoteProfiledEvents_) {
-      (*remoteProfiledEvents_).emplace_back(remoteProfiledEvents);
-    } else {
-      remoteProfiledEvents_ = {std::move(remoteProfiledEvents)};
+    void mark(std::string name, bool include_cuda = true) {
+      if (config_.state == ProfilerState::Disabled) {
+        return;
+      }
+      if (config_.state == ProfilerState::NVTX) {
+        cuda_stubs->nvtxMarkA(name.c_str());
+      } else {
+        Event evt(
+            EventKind::Mark,
+            at::StringView(std::move(name)),
+            at::RecordFunction::currentThreadId(),
+            include_cuda && config_.state == ProfilerState::CUDA);
+        evt.setNodeId(at::RecordFunction::getDefaultNodeId());
+        getEventList().record(std::move(evt));
+      }
     }
-  }
 
-  void pushRange(
-      const at::StringView& name,
-      const char* msg = "",
-      int64_t sequence_nr = -1,
-      std::vector<std::vector<int64_t>>&& shapes = {},
-      at::RecordFunctionHandle handle = 0) {
-    if (config_.state == ProfilerState::Disabled) {
-      return;
-    }
-    if (config_.state == ProfilerState::NVTX) {
-      cuda_stubs->nvtxRangePushA(getNvtxStr(
-          name, msg, sequence_nr, shapes).c_str());
-    } else {
-      Event evt(EventKind::PushRange,
-          name,
-          at::RecordFunction::currentThreadId(),
-          config_.state == ProfilerState::CUDA,
-          handle,
-          std::move(shapes),
-          at::RecordFunction::getDefaultNodeId());
-      evt.setSequenceNr(sequence_nr);
-      getEventList().record(std::move(evt));
+    void setOrAddRemoteProfiledEvents(
+        std::vector<Event>&& remoteProfiledEvents) {
+      // Lock to serialize access from multiple callback threads.
+      std::lock_guard<std::mutex> guard(state_mutex_);
+      if (remoteProfiledEvents_) {
+        (*remoteProfiledEvents_).emplace_back(remoteProfiledEvents);
+      } else {
+        remoteProfiledEvents_ = {std::move(remoteProfiledEvents)};
+      }
     }
-  }
 
-  void popRange(uint64_t thread_id, at::RecordFunctionHandle handle) {
-    if (config_.state == ProfilerState::Disabled) {
-      return;
+    void pushRange(
+        const at::StringView& name,
+        const bool record_cuda,
```

Comment:
sorry about formatting changes, pretty much the only substantial change is `record_cuda` argument here and in popRange